pkg/controller/issueapi/issue.go

```diff
@@ -192,31 +192,35 @@ func (c *Controller) HandleIssue() http.Handler {
 			}
 		}
 
-		// If we got this far, we're about to issue a code.
-		dig, err := digest.HMACUint(realm.ID, c.config.GetRateLimitConfig().HMACKey)
-		if err != nil {
-			controller.InternalError(w, r, c.h, err)
-			return
-		}
-		key := fmt.Sprintf("realm:quota:%s", dig)
-		limit, remaining, reset, ok, err := c.limiter.Take(ctx, key)
-		if err != nil {
-			logger.Errorw("failed to take from limiter", "error", err)
-			stats.Record(ctx, c.metrics.QuotaErrors.M(1))
-			c.h.RenderJSON(w, http.StatusInternalServerError, api.Errorf("failed to verify realm stats, please try again"))
-			return
-		}
-		if !ok {
-			logger.Warnw("realm has exceeded daily quota",
-				"realm", realm.ID,
-				"limit", limit,
-				"reset", reset)
-			stats.Record(ctx, c.metrics.QuotaExceeded.M(1))
-
-			if c.config.GetEnforceRealmQuotas() {
-				c.h.RenderJSON(w, http.StatusTooManyRequests, api.Errorf("exceeded realm quota"))
+		// If we got this far, we're about to issue a code - take from the limiter
+		// to ensure this is permitted.
+		if realm.AbusePreventionEnabled {
+			dig, err := digest.HMACUint(realm.ID, c.config.GetRateLimitConfig().HMACKey)
+			if err != nil {
+				controller.InternalError(w, r, c.h, err)
 				return
 			}
+			key := fmt.Sprintf("realm:quota:%s", dig)
+			limit, remaining, reset, ok, err := c.limiter.Take(ctx, key)
+			c.recordCapacity(ctx, limit, remaining)
```

Comment:
Moved this up - we were only recording capacity later, which I think would cause us to miss some data points.